File: app/src/main/java/com/me/myapplication/activity/StartDailyTestActivity.java
package com.me.myapplication.activity;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.ImageButton;
import com.me.myapplication.R;


import androidx.appcompat.app.AppCompatActivity;



public class StartDailyTestActivity extends AppCompatActivity{

    private ImageButton mButton;
    private EditText mInfo;
    String astro;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_start_daily_test);
        mButton = (ImageButton) findViewById(R.id.search_button);
        mInfo = (EditText) findViewById(R.id.start_search_info);

        mButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(StartDailyTestActivity.this, DailyTestActivity.class);
                Bundle bundle = new Bundle();
                astro = mInfo.getText().toString();
                bundle.putString("astro",astro);
                intent.putExtras(bundle);

            startActivity(intent);
            }
        });

    }
}

File: app/src/main/java/com/me/myapplication/activity/StartDailyTestActivity.kt
package com.me.myapplication.activity


import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.EditText
import android.widget.ImageButton
import androidx.appcompat.app.AppCompatActivity
import com.me.myapplication.R
import okhttp3.*


class StartDailyTestActivity : AppCompatActivity() {
    private var mButton: ImageButton? = null
    private var mInfo: EditText? = null
    var astro: String? = null


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_start_daily_test)
        mButton = findViewById<View>(R.id.search_button) as ImageButton
        mInfo = findViewById<View>(R.id.start_search_info) as EditText

        mButton!!.setOnClickListener {
            val intent = Intent(this@StartDailyTestActivity, DailyTestActivity::class.java)
            val bundle = Bundle()
            astro = mInfo!!.text.toString()

            bundle.putString("astro", astro)
            intent.putExtras(bundle)
            startActivity(intent)
        }
    }

}